src/formulation_definitions.jl

```diff
@@ -147,6 +147,37 @@ EnergyReservoirLossless is a storage formulation that does not lose energy.
 """
 struct EnergyReservoirLossless <: StoragePRAS end
 
+"""
+    InterfacePRAS <: AbstractRAFormulation
+
+InterfacePRAS produces interface entries in PRAS.
+
+If not supplied, then interfaces will be generated by Lines
+
+Subtypes must provide [`add_to_interface!`](@ref) function.
+"""
+abstract type InterfacePRAS <: AbstractRAFormulation end
+
+"""
+    AreaInterchangeInterface <: InterfacePRAS
+
+AreaInterchangeInterface produces interfaces from AreaInterchange objects
+
+Each line must have a corresponding AreaInterchange. All AreaInterchange
+objects will be consolidated for each pair of directly connected regions.
+"""
+struct AreaInterchangeInterface <: InterfacePRAS end
```

Comment:
AreaInterchangeInterface sounds weird. I'm not sure what we should change this to.